README.md

```diff
@@ -1,24 +1,157 @@
-# README
+<a name="readme-top"></a>
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+<div align="center">
 
-Things you may want to cover:
+</div>
 
-* Ruby version
 
-* System dependencies
+# 📗 Table of Contents
 
-* Configuration
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ (OPTIONAL)](#faq)
+- [📝 License](#license)
 
-* Database creation
+# 📖 vehicle booking app <a name="about-project"></a>
 
-* Database initialization
+**vehicle booking app** is a Booking app made connectively with Ruby in Rails and React for front-end [vehicle booking app](https://github.com/tan12082001/Vehicle-Booking-App-Backend.git) React frontend App in JSON format through an API endpoint.
 
-* How to run the test suite
+## 🛠 Built With <a name="built-with"></a>
 
-* Services (job queues, cache servers, search engines, etc.)
+### Tech Stack <a name="tech-stack"></a>
+  <details>
+  <summary>Client</summary>
+      <ul>
+        <li><a href="https://react.dev/">React</a></li>
+        <li><a href="https://react-redux.js.org/">React-Redux</a></li>
+        <li><a href="https://redux-toolkit.js.org/">Redux Toolkit</a></li>
+      </ul>
+  </details>
 
-* Deployment instructions
+  <details>
+  - <summary>Server</summary>
+      <ul>
+        <li><a href="https://guides.rubyonrails.org/">Ruby on Rails</a></li>
+        <li><a href="https://www.postgresql.org/docs/">PostgreSQL</a></li>
+      </ul>
+  </details>
 
-* ...
+### Key Features <a name="key-features"></a>
+
+- Rails API backend
+- React Front-end
+- Redux Toolkit used to consume the backend api
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps:
+
+### Prerequisites
+
+In order to run this project you need:
+  - A text editor of your choice ([VS Code](https://code.visualstudio.com/download) works well).
+  - [Ruby and Rails](https://gorails.com/setup/windows/10)
+  - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+
+### Setup
+
+Use the following URL to clone this project:
+
+  https://github.com/tan12082001/Vehicle-Booking-App-Backend.git
+
+### Install
+Open the terminal in the root directory of the project and run the following command to install all dependencies.
+
+    bundle install
+    npm i
+
```

Comment:
You can remove the npm i, as the dependencies will be installed through bundle install. Also let's add the instructions for generating the secret key.